File: scripts/clear-db.js
const { Pool } = require('pg');
const fs = require('fs');
const path = require('path');

// Load environment variables BEFORE accessing process.env
// Try multiple possible locations for .env files
const rootDir = path.join(__dirname, '..');
const envPaths = [
  path.join(rootDir, '.env.local'),
  path.join(rootDir, '.env'),
];

let envLoaded = false;
for (const envPath of envPaths) {
  if (fs.existsSync(envPath)) {
    require('dotenv').config({ path: envPath });
    console.log(`Loaded environment variables from ${envPath}`);
    envLoaded = true;
    break;
  }
}

if (!envLoaded) {
  console.log('No .env.local or .env file found, using default values');
}

async function clearDatabase() {
  const pool = new Pool({
    connectionString: process.env.POSTGRES_URL,
  });

  try {
    console.log('Connecting to database...');
    
    const dropSchemaSql = `
      DROP TABLE IF EXISTS users CASCADE;
      DROP TABLE IF EXISTS trainers CASCADE;
      DROP TABLE IF EXISTS sessions CASCADE;
      DROP TABLE IF EXISTS verification_tokens CASCADE;
    `;
    await pool.query(dropSchemaSql);
    
    console.log('Database cleared successfully!');
  } catch (error) {
    console.error('Error clearing database:', error);
    process.exit(1);
  } finally {
    await pool.end();
  }
}

clearDatabase();

File: scripts/clear-db.py
# /// script
# requires-python = ">=3.11"
# dependencies = ["psycopg2-binary", "python-dotenv"]
# ///

import argparse
import os
import sys
from pathlib import Path

import psycopg2
from dotenv import load_dotenv


def parse_args():
    parser = argparse.ArgumentParser(description="Drop all application tables from the database.")
    parser.add_argument("--env", type=Path, required=True, help="Path to the .env file")
    return parser.parse_args()


def main():
    args = parse_args()

    if not args.env.exists():
        print(f"Error: {args.env} not found", file=sys.stderr)
        sys.exit(1)

    load_dotenv(args.env)
    print(f"Loaded {args.env}")

    conn = psycopg2.connect(os.environ["POSTGRES_URL"])
    try:
        with conn:
            cur = conn.cursor()
            cur.execute("""
                DROP TABLE IF EXISTS rate_limits CASCADE;
                DROP TABLE IF EXISTS verifications CASCADE;
                DROP TABLE IF EXISTS accounts CASCADE;
                DROP TABLE IF EXISTS sessions CASCADE;
                DROP TABLE IF EXISTS trainers CASCADE;
                DROP TABLE IF EXISTS users CASCADE;
            """)
        print("Database cleared successfully!")
    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)
    finally:
        conn.close()


if __name__ == "__main__":
    main()
